pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4295,202 +4416,122 @@ int BPF_KPROBE(cgroup_bpf_run_filter_skb)
+            if (nethdrs->iphdrs.ipv6hdr.version != 6) // IPv6
+                return 1;
 
-static __always_inline u32 cgroup_skb_submit(void *map,
-                                             struct __sk_buff *ctx,
-                                             net_event_context_t *neteventctx,
-                                             u32 event_type,
-                                             u32 size)
-{
-    u64 flags = BPF_F_CURRENT_CPU;
+            proto = nethdrs->iphdrs.ipv6hdr.nexthdr;
+            switch (proto) {
+                case IPPROTO_TCP:
+                case IPPROTO_UDP:
+                case IPPROTO_ICMPV6:
+                    break;
+                default:
+                    return 1; // ignore other protocols
+            }
 
-    size = size > FULL ? FULL : size;
-    switch (size) {
-        case HEADERS:
-            size = neteventctx->md.header_size;
-            break;
-        case FULL:
-            size = ctx->len;
+            // add IPv6 header items to indexer
+            __builtin_memcpy(&indexer.ip_saddr.in6_u, &nethdrs->iphdrs.ipv6hdr.saddr.in6_u, 4 * sizeof(u32));
+            __builtin_memcpy(&indexer.ip_daddr.in6_u, &nethdrs->iphdrs.ipv6hdr.daddr.in6_u, 4 * sizeof(u32));
             break;
+
         default:
-            size += neteventctx->md.header_size;       // add headers size
-            size = size > ctx->len ? ctx->len : size;  // check limits
-            break;
+            return 1;
     }
 
-    flags |= (u64) size << 32;
-    neteventctx->bytes = size + sizeof(u32);
-
-    // set the event type before submitting event
-    neteventctx->eventctx.eventid = event_type;
-
-    return bpf_perf_event_output(ctx,
-                                 map,
-                                 flags,
-                                 neteventctx,
-                                 sizeof_net_event_context_t());
-}
-
-#define cgroup_skb_submit_event(a,b,c,d) cgroup_skb_submit(&events,a,b,c,d)
-
-static __always_inline u32 cgroup_skb_capture_event(struct __sk_buff *ctx,
-                                                    net_event_context_t *neteventctx,
-                                                    u32 event_type)
-{
-    int zero = 0;
-
-    // pick network config map to know requested capture length
-    netconfig_entry_t *nc = bpf_map_lookup_elem(&netconfig_map, &zero);
-    if (nc == NULL)
-        return 0;
+    //
+    // LINK CONTENT INDEXER TO EVENT CONTEXT
+    //
 
-    return cgroup_skb_submit(&net_cap_events,
-                             ctx,
-                             neteventctx,
-                             event_type,
-                             nc->capture_length);
-}
+    neteventctx.bytes = 0; // event arg size: no payload by default (changed inside skb prog)
 
-// capture packet a single time (if passing through multiple protocols being submitted to userland)
+    // TODO: log collisions
```

Comment:
I'll have to implement logs for network errors. If I had that placed before, I would have understood much sooner what was going on with the code (map key conflicts and missing logic).

Reply:
@rafaeldtinoco